Sources/BrowserServicesKit/Autofill/AutofillPixelReporter.swift

```diff
@@ -0,0 +1,225 @@
+
+    enum Parameter {
+        static let countBucket = "count_bucket"
+    }
+}
+
+public final class AutofillPixelReporter {
+
+    enum Keys {
+        static let autofillSearchDauDateKey = "com.duckduckgo.app.autofill.SearchDauDate"
+        static let autofillFillDateKey = "com.duckduckgo.app.autofill.FillDate"
+        static let autofillOnboardedUserKey = "com.duckduckgo.app.autofill.OnboardedUser"
+    }
+
+    enum BucketName: String {
+        case none
+        case few
+        case some
+        case many
+        case lots
+    }
+
+    private enum EventType {
+        case fill
+        case searchDAU
+    }
+
+    private let userDefaults: UserDefaults
+    private let eventMapping: EventMapping<AutofillPixelEvent>
+    private var secureVault: (any AutofillSecureVault)?
+    private var reporter: SecureVaultReporting?
+    private var installDate: Date?
+
+    private var autofillSearchDauDate: Date? { userDefaults.object(forKey: Keys.autofillSearchDauDateKey) as? Date ?? .distantPast }
+    private var autofillFillDate: Date? { userDefaults.object(forKey: Keys.autofillFillDateKey) as? Date ?? .distantPast }
+    private var autofillOnboardedUser: Bool { userDefaults.object(forKey: Keys.autofillOnboardedUserKey) as? Bool ?? false }
+
+    public init(userDefaults: UserDefaults,
+                eventMapping: EventMapping<AutofillPixelEvent>,
+                secureVault: (any AutofillSecureVault)? = nil,
+                reporter: SecureVaultReporting? = nil,
+                installDate: Date? = nil
+    ) {
+        self.eventMapping = eventMapping
+        self.userDefaults = userDefaults
+        self.secureVault = secureVault
+        self.reporter = reporter
+        self.installDate = installDate
+
+        createNotificationObservers()
+    }
+
+    public func resetStoreDefaults() {
+        userDefaults.set(Date.distantPast, forKey: Keys.autofillSearchDauDateKey)
+        userDefaults.set(Date.distantPast, forKey: Keys.autofillFillDateKey)
+        userDefaults.set(false, forKey: Keys.autofillOnboardedUserKey)
+    }
+
+    private func createNotificationObservers() {
+        NotificationCenter.default.addObserver(self, selector: #selector(didReceiveSearchDAU), name: .searchDAU, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(didReceiveFillEvent), name: .autofillFillEvent, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(didReceiveSaveEvent), name: .autofillSaveEvent, object: nil)
+    }
+
+    @objc
+    private func didReceiveSearchDAU() {
+        guard let autofillSearchDauDate = autofillSearchDauDate, !Date.isSameDay(Date(), autofillSearchDauDate) else {
+            return
+        }
+
+        userDefaults.set(Date(), forKey: Keys.autofillSearchDauDateKey)
+
+        firePixelsFor(.searchDAU)
+    }
+
+    @objc
+    private func didReceiveFillEvent() {
+        os_log(.debug, "Autofill fill event fired")
```

Comment:
I put this temporary log in to make testing easier; I'll be removing before merging